src/main/java/com/jwplayer/jwplatform/rest/HttpCalls.java

```diff
@@ -0,0 +1,158 @@
+	   * @throws JWPlatformException - an exception occurred during encoding
+	   */
+	  private static String buildRequestUrl(
+	      final String path, final Map<String, String> params)
+	      throws JWPlatformException {
+	    final TreeMap<String, String> orderedParams = new TreeMap<>(params);
+	    orderedParams.put("api_format", "json");
+	    orderedParams.put("api_nonce", getRandomNonce());
+	    orderedParams.put("api_timestamp", getCurrentUnixTimestampInSeconds());
+
+	    final StringBuilder encodedParams = new StringBuilder();
+	    for (final String param : orderedParams.keySet()) {
+	      if (encodedParams.length() != 0) {
+	        encodedParams.append("&");
+	      }
+	      final String encodedValue = encodeStringForJWPlatformAPI(orderedParams.get(param));
+	      encodedParams.append(param).append("=").append(encodedValue);
+	    }
+
+	    return path + "?" + encodedParams.toString();
+	  }
+	  
+	  /**
+	   * URL encodes a {@code String}, then modifies it to be compliant with the JW Platform API.
+	   *
+	   * @param stringToEncode - the {@code String} to be URL Encoded
+	   * @return - JW Platform API compliant encoded {@code String}
+	   * @throws JWPlatformException - an exception occurred trying to encode the requested
+	   *     {@code String}
+	   */
+	  private static String encodeStringForJWPlatformAPI(final String stringToEncode)
+	          throws JWPlatformException {
+	    try {
+	      final String encodedValue = URLEncoder.encode(stringToEncode, "utf-8");
+
+	      // string replacements to align with the API
+	      return encodedValue.replace("%7E", "~").replace("*", "%2A").replace("+", "%20");
+	    } catch (final UnsupportedEncodingException e) {
+	      throw new JWPlatformUnknownException(
+	              String.format("Exception thrown encoding URL parameter %s", e.toString()));
+	    }
+	  }
+	  
+	  /**
+	   * Check if the API response is an non-200. If so, throw the
+	   * appropriate JWPlatformException exception based on the
+	   * error message.
+	   *
+	   * @param statusCode - the response status code
+	   * @param response - a {@code JSONObject} object with the API response block
+	   * @throws JWPlatformException - API returned an exception
+	   */
+	  private static void checkForNon200Response(final int statusCode, final JSONObject response)
+	          throws JWPlatformException {
+	    if (statusCode != 200 && statusCode != 201) {
+	      try {
+	        MediaAPIExceptionFactory.throwJWPlatformException(
+	                StringUtils.stripEnd(response.getString("code"), "Error"), response.toString());
+	      } catch (final JSONException e) {
+	        throw new JWPlatformUnknownException(
+	                String.format("Unknown JSONException thrown: %s", e.toString()));
+	      }
+	    }
+	  }
+	  
+	  /**
+	   * Returns the current unix timestamp in seconds.
+	   *
+	   * @return - unix timestamp in seconds
+	   */
+	  private static String getCurrentUnixTimestampInSeconds() {
```

Comment:
you can move this and the below function to a utils class and share it between this class and the legacy v1 since they both use it.